src/main/java/com/faforever/api/security/FafUserDetailsService.java

```diff
@@ -23,6 +23,9 @@ public FafUserDetailsService(UserRepository userRepository) {
 
   @Override
   public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
+    if (userRepository.findOneByLoginIgnoreCase(username) == null) {
```

Comment:
`userRepository.findOneByLoginIgnoreCase(username)` is called twice within this method, which potentially causes two database calls.

Assign the result to a variable instead (Select statement -> CTRL+ALT+V) and use the variable instead. This way there is only one databse call.